internal/server/middleware.go

```diff
@@ -113,6 +116,27 @@ func withDBTxn(ctx context.Context, db *gorm.DB, fn func(tx *gorm.DB)) {
 func unauthenticatedMiddleware(db *gorm.DB) gin.HandlerFunc {
 	return func(c *gin.Context) {
 		withDBTxn(c.Request.Context(), db, func(tx *gorm.DB) {
+			var orgID uid.ID
```

Comment:
I think this should set the default org in the case no Infra-Org header is found

Reply:
moved this to a separate PR